fe/fe-core/src/main/java/org/apache/doris/analysis/AlterSqlBlockRuleStmt.java

```diff
@@ -0,0 +1,84 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package org.apache.doris.analysis;
+
+import org.apache.doris.block.SqlBlockRule;
+import org.apache.doris.catalog.Catalog;
+import org.apache.doris.common.AnalysisException;
+import org.apache.doris.common.UserException;
+
+import org.apache.commons.lang3.StringUtils;
+
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+public class AlterSqlBlockRuleStmt extends DdlStmt {
+
+    private final String ruleName;
+
+    private String user;
+
+    private String sql;
+
+    private Boolean enable;
+
+    private final Map<String, String> properties;
+
+    public AlterSqlBlockRuleStmt(String ruleName, Map<String, String> properties) {
+        this.ruleName = ruleName;
+        this.properties = properties;
+    }
+
+    @Override
+    public void analyze(Analyzer analyzer) throws UserException {
+        super.analyze(analyzer);
+        // check properties
+        checkProperties(properties);
```

Comment:
privilege check is always in `analyze` phase. 
And I think only check `admin` priv is enough

Reply:
`checkProperties` whether needed？I referred to `AlterRoutineLoadStmt`

Reply:
AlterRoutineLoadStmt is a little bit specially because it need table info, which not contains in alter stmt.
But for this stmt, we can check ADMIN priv here.

Of cause `checkProperties` is needed.